src/Microsoft.ML.CpuMath/CpuMathUtils.netcoreapp.cs

```diff
@@ -9,19 +9,199 @@ namespace Microsoft.ML.Runtime.Internal.CpuMath
 {
     public static partial class CpuMathUtils
     {
+        public static void MatTimesSrc(bool tran, bool add, AlignedArray mat, AlignedArray src, AlignedArray dst, int crun)
+        {
+            Contracts.Assert(mat.Size == dst.Size * src.Size);
+
+            if (Sse.IsSupported)
+            {
+                if (!tran)
+                {
+                    Contracts.Assert(crun >= 0);
```

Comment:
since this assert is in all the paths, we can move it out to the common path, right?